src/Servers/Kestrel/Core/src/KestrelServerOptions.cs

```diff
@@ -278,6 +282,19 @@ internal void ApplyDefaultCertificate(HttpsConnectionAdapterOptions httpsOptions
         httpsOptions.ServerCertificate = DevelopmentCertificate;
     }
 
+    internal void EnableHttpsConfiguration()
+    {
+        var HttpsConfigurationService = ApplicationServices.GetRequiredService<IHttpsConfigurationService>();
+
+        if (!HttpsConfigurationService.IsInitialized)
+        {
+            var hostEnvironment = ApplicationServices.GetRequiredService<IHostEnvironment>();
+            var logger = ApplicationServices.GetRequiredService<ILogger<KestrelServer>>();
+            var httpsLogger = ApplicationServices.GetRequiredService<ILogger<HttpsConnectionMiddleware>>();
+            HttpsConfigurationService.Initialize(hostEnvironment, logger, httpsLogger);
+        }
```

Comment:
Usually, there's a single call to some entry point service that resolved dependencies. This feels *very* manual. Is there a way to avoid these checks to `IsInitialized` or does that happen opportunistically in various places?

Reply:
This is the pattern I came up with for making it possible to light up this functionality _either_ during DI setup or afterwards (esp when calling `ListenOptions.UseHttps`).  Alternatives welcome.

Reply:
To elaborate a bit, there's always an `IHttpsConfigurationService`, but all of its methods throw unless you either register an `HttpsConfigurationService.IInitializer` (as in `UseHttpsConfiguration`) or explicitly invoke `Initialize` (as here, in service of `ListenOptions.UseHttps`).